### PR TITLE
ci: Switch generic linux_job to use conda images

### PR DESCRIPTION
**Pull request by seemethere**
conda images are better since conda automatically comes with all of the python versions and people tend to like conda better for test environments

Signed-off-by: Eli Uriegas <eliuriegas@fb.com>

### CODE REVIEW

File: .github/workflows/test_linux_job.yml [hunk 1]
```diff
@@ -17,7 +17,8 @@ jobs:
       gpu-arch-type: cpu
       gpu-arch-version: ""
       script: |
-        export PATH="/opt/python/cp38-cp38/bin:${PATH}"
+        conda create -y -n test python=3.8
+        conda activate test
```
Comment (izaitsevfb) [inline .github/workflows/test_linux_job.yml:21]: The [test job](https://github.com/pytorch/test-infra/actions/runs/3236769746/jobs/5303019152#step:8:169) seems to have some issues:
```

Retrieving notices: ...working... done
+ conda activate test

CommandNotFoundError: Your shell has not been properly configured to use 'conda activate'.
To initialize your shell, run

    $ conda init <SHELL_NAME>
```

Sees like the shell is not properly configured to use conda?

**Comment by colin2328** [comment]
does conda activate in GHA work? according to https://python.plainenglish.io/eureka-i-finally-managed-to-activate-my-conda-environment-in-my-ci-pipeline-a8b578c21f03 this was hard to get working successfully

**Comment by seemethere** [comment]
> does conda activate in GHA work? according to [python.plainenglish.io/eureka-i-finally-managed-to-activate-my-conda-environment-in-my-ci-pipeline-a8b578c21f03](https://python.plainenglish.io/eureka-i-finally-managed-to-activate-my-conda-environment-in-my-ci-pipeline-a8b578c21f03) this was hard to get working successfully

Actually going to change this to utilize `conda shell.bash hook` instead. Since we only need to activate it once it'll probably make it easier

**Review by izaitsevfb — APPROVED** [review]
LGTM

**Pull request merged**